### PR TITLE
refactor: Cleanup user state

### PR DESCRIPTION
**Pull request by atomrc**
## Description

We currently have a weird relationship between the `UserState`, the `TeamState` and the `ConversationState`. 
The `UserState` get some of its state from those other two. Which make things hard to follow and understand. 

This PR brings back the single source of truth to find information about users and adds documentation to explain what each observable represents

## Checklist

- [x] PR has been self reviewed by the author;
- [x] Hard-to-understand areas of the code have been commented;
- [x] If it is a core feature, unit tests have been added;

### CODE REVIEW

File: src/script/team/TeamState.ts [hunk 1]
```diff
@@ -82,10 +84,6 @@ export class TeamState {
 
     this.supportsLegalHold = ko.observable(false);
 
-    this.userState.isTeam = this.isTeam;
-    this.userState.teamMembers = this.teamMembers;
-    this.userState.teamUsers = this.teamUsers;
-
```
Comment (atomrc) [inline src/script/team/TeamState.ts:88]: Those are weird assignement from one observable to another one. Let's use the single source of truth there and only consume the `TeamState` when we need to know who are the team members, for example

File: src/script/user/UserState.ts [hunk 2]
```diff
@@ -53,16 +46,5 @@ export class UserState {
           .sort(sortUsersByPriority);
       })
       .extend({rateLimit: TIME_IN_MILLIS.SECOND});
-
-    this.isTeam = ko.observable();
-    this.teamMembers = ko.pureComputed((): User[] => []);
-    this.teamUsers = ko.pureComputed((): User[] => []);
```
Comment (atomrc) [inline src/script/user/UserState.ts:59]: all those were actually reassigned later at runtime (not great for readability)

**Review by aweiss-dev — APPROVED** [review]

**Review by V-Gira — APPROVED** [review]

**Review by aweiss-dev — APPROVED** [review]
